src/lib/tauri/http/Response.ts

```diff
@@ -0,0 +1,51 @@
+import * as http from '@tauri-apps/api/http';
+import { either, flow, iots } from '@code-expert/prelude';
+import { HttpError, httpError } from './HttpError';
+
+const ResponseError = iots.strict({
+  statusCode: iots.number,
+  error: iots.string,
+  message: iots.string,
+});
+type ResponseError = iots.TypeOf<typeof ResponseError>;
+
+const Response = iots.union([
+  iots.strict({
+    ok: iots.literal(true),
+    data: iots.unknown,
+  }),
+  iots.strict({
+    ok: iots.literal(false),
+    data: ResponseError,
+  }),
+]);
+
+export const parseResponse: <A>(
+  codec: iots.Decoder<unknown, A>,
+) => (r: http.Response<unknown>) => either.Either<HttpError, A> = (codec) =>
+  flow(
+    Response.decode,
+    either.mapLeft(iots.formatValidationErrors),
+    either.getOrThrow(
+      (errs) =>
+        new Error(`Fetch response did not match Tauri’s "Response" type': ${errs.join('; ')}`),
+    ),
+    (res) => (res.ok ? either.right(res.data) : either.left(toHttpError(res.data))),
+    either.map(
+      flow(
+        codec.decode,
+        either.mapLeft(iots.formatValidationErrors),
+        either.getOrThrow(
+          (errs) => new Error(`Response payload did not match codec: ${errs.join('; ')}`),
+        ),
+      ),
+    ),
+  );
+
+// -------------------------------------------------------------------------------------------------
+
+const toHttpError = ({ statusCode, message }: ResponseError): HttpError => {
+  if (500 <= statusCode && statusCode < 600) return httpError.serverError({ statusCode, message });
```

Comment:
```suggestion
  if (statusCode >= 500 && statusCode < 600) return httpError.serverError({ statusCode, message });
```
I guess you're mimicking python's `500 <= statusCode < 600`, but I think it's more legible like this

Reply:
Yes, I wanted to express the lower and higher boundary, but I have no strong feelings about this, so I flipped it.